fishtank/src/cluster.ts

```diff
@@ -36,4 +36,19 @@ export class Cluster {
     })
     return new Node(this, name)
   }
+
+  async teardown(): Promise<void> {
+    // Remove containers
+    const removeContainers = []
+    const namePrefix = this.containerName('')
+    for (const container of await this.backend.list()) {
+      if (container.name.startsWith(namePrefix)) {
```

Comment:
If we are teardown the cluster, we should remove all the containers returned by `list()`. Why there is extra filtering logic? 

If we are teardown a network, the filtering logic seems to ensure we only remove the container for that network, which will make sense. But the name prefix value should be from `this.network`. It looks a little hacky to pass in `''` to get container name prefix. You can create a new helper function to get container name prefix. Currently this code make it a tribute knowledge, it will not be good to new developers contribute to the code base.

Reply:
> If we are teardown the cluster, we should remove all the containers returned by list(). Why there is extra filtering logic?

Because one might be running other workloads in their Docker daemon. We don't want to nuke everything a user is running, only the resources created by this simulator.

Reply:
Make sense! can you use cluster name to identify the containers 
```
if (container.name.startsWith(this.name))
```